guides/v3.5.0/tutorial/service.md

```diff
@@ -108,56 +61,59 @@ Let's get started creating our service by generating it through Ember CLI,
 which will create the service file, as well as a unit test for it.
 
 ```bash
-ember g service maps
+ember g service map-element
 ```
 
 Now implement the service as follows.
+The main API will be an [async function](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function) called `getMapElement`,
+that returns a HTML element containing a rendered map for the given location.
+
 Note that we check if a map already exists for the given location and use that one,
-otherwise we call a Google Maps utility to create one.
+otherwise we will create a new HTML element and call our Leaflet map service to render a map to it.
 
-```javascript {data-filename="app/services/maps.js" data-diff="+2,+3,+5,+9,+10,+11,+12,+13,+14,+15,+16,+17,+19,+20,+21,+22,+23,+24,+25,+26,+27,+28,+30,+31,+32,+33,+34"}
-import Service from '@ember/service';
+```javascript {data-filename="app/services/map-element.js" data-diff="+1,+2,+3,+4,+6,+8,+9,+11,+12,+13,+14,+15,+16,+18,+19,+20,+21,+22,+23,+24,+25,+26,+27,+28,+30,+31,+32,+33,+34,+35"}
 import { camelize } from '@ember/string';
-import EmberObject from '@ember/object';
-
-import MapUtil from '../utils/google-maps';
+import { get, set } from '@ember/object';
+import Service from '@ember/service';
+import { inject as service } from '@ember/service';
 
 export default Service.extend({
 
+  geocode: service(),
+  map: service(),
+
   init() {
-    this._super(...arguments);
     if (!this.cachedMaps) {
-      this.set('cachedMaps', EmberObject.create());
-    }
-    if (!this.mapUtil) {
-      this.set('mapUtil', MapUtil.create());
+      set(this, 'cachedMaps', {});
     }
+    this._super(...arguments);
   },
 
-  getMapElement(location) {
+  async getMapElement(location) {
     let camelizedLocation = camelize(location);
-    let element = this.get(`cachedMaps.${camelizedLocation}`);
+    let element = get(this, `cachedMaps.${camelizedLocation}`);
```

Comment:
I think I can get away with accessing it through `this.chachedMaps[camelizedLocation]` instead.  What do you think?

Reply:
Fixt